src/psyclone/psyir/nodes/loop.py

```diff
@@ -362,19 +395,30 @@ def kernel(self, kern):
         self._kern = kern
 
     @property
-    def variable_name(self):
+    def variable(self):
         '''
-        :returns: the name of the control variable for this loop.
-        :rtype: str
+        :returns: a reference to the control variable for this loop.
+        :rtype: :py:class:`psyclone.psyir.symbols.DataSymbol`
+        '''
+        return self._variable
+
+    @variable.setter
+    def variable(self, var):
+        '''
+        Setter for the variable associated with this loop.
+
+        :param var: the control variable reference.
+        :type var: :py:class:`psyclone.psyir.symbols.DataSymbol`
+
         '''
-        return self._variable_name
+        self._variable = var
```

Comment:
This should call `_check_variable` at least to avoid adding references, but also to check it is a scalar integer.

Reply:
Yes, quite right. I missed that.